File: Data Structures & Algorithms/is-palindrome/submission-7.kt
class Solution {
    fun isPalindrome(s: String): Boolean {
        var left = 0
        var right = s.length - 1
        
        while (left < right) {
            while (!s[left].isLetterOrDigit() && left < right) {
                left++
            }
            while (!s[right].isLetterOrDigit() && left < right) {
                right--
            }
            
            if (s[left].lowercase() != s[right].lowercase()) {
                return false
            }
            
            left++
            right--
        }

        return true
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-7.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        left, right = 0, 1

        result = 0

        while right < len(prices):
            buy = prices[left]
            sell = prices[right]
            diff = sell - buy

            result = max(diff, result)

            if sell < buy:
                left = right

            right += 1


        return result